assets/js/alerts.js

```diff
@@ -0,0 +1,228 @@
+  }
+  return await response.json()
+}
+
+//add current filters to Url params
+function updateURLParams(filters) {
+  const params = new URLSearchParams()
+  if (filters.application.length) params.set('application', filters.application)
+  if (filters.environment) params.set('environment', filters.environment)
+  if (filters.namespace) params.set('namespace', filters.namespace)
+  if (filters.severity) params.set('severity', filters.severity)
+  history.replaceState(null, '', '?' + params.toString())
+}
+// function checks url params for applied filters and builds filter object
+function getFiltersFromURL() {
+  const params = new URLSearchParams(location.search)
+  return {
+    application: params.get('application') || '',
+    environment: params.get('environment') || '',
+    namespace: params.get('namespace') || '',
+    severity: params.get('severity') || '',
+  }
+}
+
+//append tabledata to the #statusTable
+async function populateAlertTable(alerts) {
+  try {
+    $('#statusRows').empty()
+    alerts.forEach(alert => {
+      //create links for alert urls
+      const dashboardLink = alert.annotations.dashboard_url
+        ? `<a href="${alert.annotations.dashboard_url}" class="statusTileHealth" target="_blank">View</a>`
+        : 'N/A'
+      const runbookLink = alert.annotations.runbook_url
+        ? `<a href="${alert.annotations.runbook_url}" class="statusTileHealth" target="_blank">View</a>`
+        : 'N/A'
+      const generatorLink = alert.generatorURL
+        ? `<a href="${alert.generatorURL}" class="statusTileHealth" target="_blank">View</a>`
+        : 'N/A'
+      $('#statusRows')
+        .append(`<tr data-alert-name="${alert.labels.application}" data-environment="${alert.labels.application}" data-environment-type="${alert.labels.environment}" id="tile-${alert.labels.application}-${alert.labels.environment}">
+          <td>${alert.labels.alertname}</td>
+          <td>${alert.annotations.message} </td>
+          <td>${dashboardLink}</td>
+          <td>${runbookLink} </td>
+          <td>${generatorLink}</td>
+        </tr>`)
+    })
+  } catch (e) {
+    console.error(e)
+  }
+}
+
+function getOptions(data, key) {
+  const set = new Set(data.map(a => a.labels[`${key}`]))
+  return Array.from(set).sort()
+}
+
+function removeOptions(selectElement) {
+  var i,
+    L = selectElement.options.length - 1
+  for (i = L; i >= 0; i--) {
+    selectElement.remove(i)
+  }
+  const opt = document.createElement('option')
+  opt.value = ''
+  opt.textContent = ''
+  selectElement.appendChild(opt)
+}
+
+function renderDropdown(select, options, selectedValue, key) {
+  //clear dropDown options to prevent duplicates
+  removeOptions(select)
+  if (selectedValue != '') {
+    //append selected option
+    options.forEach(option => {
+      const opt = document.createElement('option')
+      opt.value = option
+      opt.textContent = option
+      if (option === selectedValue) opt.selected = true
```

Comment:
Also try to make sure to always use `{}` the linter would usually pick up I think.